File: materials/2022-2023/11a/shapes/triangle.cpp
#include "triangle.hh"
#include <math.h>

float triangle::perimeter() const
{
    return getDistance(a, b) + getDistance(b, c) + getDistance(c, a);
}

float triangle::area() const
{
    float halfPerimeter = perimeter() / 2;

    return sqrt((halfPerimeter - getDistance(a, b)) * (halfPerimeter - getDistance(b, c)) * (halfPerimeter - getDistance(c, a)));
}

File: materials/2022-2023/11a/2022-11-04-shapes/Shapes/src/Triangle.java
public class Triangle implements Shape, HasHeight {
    private Point a;
    private Point b;
    private Point c;

    public Triangle(Point a, Point b, Point c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }
    public double getPerimeter()
    {
        return Point.getSide(a, b) + Point.getSide(a, c) + Point.getSide(b, c);
    }
    public double getArea()
    {
        double halfPerimeter = getPerimeter() / 2;

        return Math.sqrt(halfPerimeter *
                (halfPerimeter - Point.getSide(a, b) *
                        (halfPerimeter - Point.getSide(a, c) *
                                (halfPerimeter - Point.getSide(b, c)))));
    }
    public double getHeight()
    {
        PointHoader temp = Point.max(a, b, c);

        return getArea() / 2 * Point.getSide(temp.getOtherPoints()[0], temp.getOtherPoints()[1]);
    }
}
